### PR TITLE
CP-49002: Write the best practice develop document

### PR DESCRIPTION
**Pull request by xueqingz**

### CODE REVIEW

File: Develop.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# Project Development Documentation
+
+## Overview
+
+This is the best practice document for XenServer terraform provider development, target to make contributors understand how to make changes, run local checks or tests, trigger automatic XenRT tests, raise and merge PRs.
```
Comment (initisan) [inline Develop.md:5]: Do we need to mention that XenRT now is only for XenServer Internal use?

File: Develop.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+Before push your commit, suggest to run below checks and tests to confirm the code quality first.
+
+1. Format code with `gofmt -w -l xenserver/*.go` and run `golangci-lint` to make sure no Go lint error.
+
+```shell
+sudo docker run -it -v $(pwd):/app -w /app golangci/golangci-lint bash
+golangci-lint run --config=/app/.golangci.yml
+```
+
+2. Run and pass the full suite of acceptance tests with `make testacc`.
```
Comment (initisan) [inline Develop.md:46]: In order to pass this test, we still need to setup a xs instance. is that need to mentioned here, or this `develop.md` only for internal use?

**Review by initisan — APPROVED** [review]

File: Develop.md (hunk 2 above)
Comment (xueqingz) [reply Develop.md:46]: yes, we need to mention here, this document is for both internal and community use.

File: Develop.md (hunk 1 above)
Comment (xueqingz) [reply Develop.md:5]: I think the the different develop process already be covered in "## Development Process", include the XenRT. No need to mention here.

**Pull request closed without merging**